core/metamodel/src/main/java/org/apache/isis/core/metamodel/facets/object/domainobjectlayout/CssClassFaFacetForDomainObjectLayoutAnnotation.java

```diff
@@ -0,0 +1,42 @@
+/* Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ * 
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License. */
+package org.apache.isis.core.metamodel.facets.object.domainobjectlayout;
+
+import org.apache.isis.applib.annotation.CssClassFaPosition;
+import org.apache.isis.applib.annotation.DomainObjectLayout;
+import org.apache.isis.core.metamodel.facetapi.FacetHolder;
+import org.apache.isis.core.metamodel.facets.members.cssclassfa.CssClassFaFacet;
+import org.apache.isis.core.metamodel.facets.members.cssclassfa.CssClassFaFacetAbstract;
+
+import com.google.common.base.Strings;
+
+public class CssClassFaFacetForDomainObjectLayoutAnnotation extends CssClassFaFacetAbstract {
```

Comment:
Hi Martin,

should I try to merge them?

I have to admit, that most of the changes I did were mere monky-see/monkey-do without much design.

Best regards
Jörg

Von: Martin Grigorov [mailto:notifications@github.com]
Gesendet: Dienstag, 14. April 2015 21:44
An: apache/isis
Cc: Rade, Joerg / Kuehne + Nagel / Ham GI-PS
Betreff: Re: [isis] Isis 1085 css class fa (#29)


In core/metamodel/src/main/java/org/apache/isis/core/metamodel/facets/object/domainobjectlayout/CssClassFaFacetForDomainObjectLayoutAnnotation.java<https://github.com/apache/isis/pull/29#discussion_r28362949>:

> + * software distributed under the License is distributed on an

> + * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY

> + * KIND, either express or implied. See the License for the

> + * specific language governing permissions and limitations

> + * under the License. */

> +package org.apache.isis.core.metamodel.facets.object.domainobjectlayout;

> +

> +import org.apache.isis.applib.annotation.CssClassFaPosition;

> +import org.apache.isis.applib.annotation.DomainObjectLayout;

> +import org.apache.isis.core.metamodel.facetapi.FacetHolder;

> +import org.apache.isis.core.metamodel.facets.members.cssclassfa.CssClassFaFacet;

> +import org.apache.isis.core.metamodel.facets.members.cssclassfa.CssClassFaFacetAbstract;

> +

> +import com.google.common.base.Strings;

> +

> +public class CssClassFaFacetForDomainObjectLayoutAnnotation extends CssClassFaFacetAbstract {

This class also looks very similar to CssClassFaFacetForDomainObjectLayoutFactory.
I'll have to check the current code to see why both of them are needed.

—
Reply to this email directly or view it on GitHub<https://github.com/apache/isis/pull/29/files#r28362949>.

Kühne + Nagel (AG & Co.) KG
Rechtsform: Kommanditgesellschaft, Bremen HRA 21928, USt-IdNr.: DE 812773878.
Geschäftsleitung Kühne + Nagel (AG & Co.) KG: Reiner Heiken (Vors.), Dirk Blesius, Martin Brinkmann, Holger Ketz, Jan-Hendrik Köstergarten, Christian Solf, Lars Wedel, Jens Wollesen.
Persönlich haftende Gesellschafterin: Kühne & Nagel A.G., Rechtsform: Aktiengesellschaft nach luxemburgischem Recht, HR-Nr.: B 18745, Geschäftsführendes Verwaltungsratsmitglied: Karl Gernandt.
Geschäftsleitung Region Westeuropa: Yngve Ruud (Vors.), Richard Huhn, Björn Johansson, Jan Kunze, Bruno Mang, Stefan Paul, Holger Ketz, Dominic Edmonds.

Wir arbeiten ausschließlich auf Grundlage der Allgemeinen Deutschen Spediteursbedingungen (ADSp), jeweils neuester Fassung. Wir verweisen insbesondere auf die vom Gesetz abweichenden Haftungsbeschränkungen von Ziffer 23 und 24 ADSp. Den vollständigen Text der ADSp übersenden wir Ihnen gerne auf Anfrage und können Sie auch unter http://www.kuehne-nagel.com einsehen. Ergänzend wird vereinbart, dass (1) Ziffer 27 ADSp im Rahmen internationaler Übereinkommen weder unsere Haftung noch die Zurechnung des Verschuldens von Leuten und sonstigen Dritten zu Gunsten des Auftraggebers erweitert, und (2) wir in den im deutschen Seehandelsrecht aufgeführten Fällen des nautischen Verschuldens oder Feuer an Bord nur für eigenes Verschulden und (3) im Sinne der CMNI genannten Voraussetzungen nicht für nautisches Verschulden, Feuer an Bord oder Mängel des Schiffes haften.